core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -406,10 +510,13 @@ class LogManager(val logDirs: Array[File],
    * Create a log for the given topic and the given partition
    * If the log already exists, just return a copy of the existing log
    */
-  def createLog(topicPartition: TopicPartition, config: LogConfig): Log = {
+  def createLog(topicPartition: TopicPartition, config: LogConfig, isNew: Boolean = false): Log = {
```

Comment:
Sure. I added the following Java doc:

```
  /**
   * If the log already exists, just return a copy of the existing log
   * Otherwise if isNew=true or if there is no offline log directory, create a log for the given topic and the given partition
   * Otherwise throw KafkaStorageException
   *
   * @param isNew Whether the replica should have existed on the broker or not
   * @throws KafkaStorageException if isNew=false, log is not found in the cache and there is offline log directory on the broker
   */
```